### PR TITLE
Add `TOML` grammar

### PR DESCRIPTION
**Pull request by mishazharov**
This PR adds the `TOML` grammar

### CODE REVIEW

**Comment by meain** [comment]
Hey, the CI had a few issues which I have now fixed in master. Could you rebase your branch on top of latest master?

**Comment by mishazharov** [comment]
Thanks for taking a look at the CI. I've rebased this branch now

File: tree-sitter-langs.el [hunk 1]
```diff
@@ -138,6 +138,7 @@ See `tree-sitter-langs-repos'."
                 (rustic-mode     . rust)
                 (scala-mode      . scala)
                 (swift-mode      . swift)
+                (toml-mode       . toml)
```
Comment (mishazharov) [inline tree-sitter-langs.el:141]: I had a quick question, is `toml-mode` appropriate here? Not sure what value this should be

Comment (meain) [reply tree-sitter-langs.el:141]: Yup, this looks good. This is a mapping from the major mode to the lang.

File: tree-sitter-langs-build.el [hunk 2]
```diff
@@ -245,7 +245,7 @@ infrequent (grammar-only changes). It is different from the version of
   "List of suffixes for shared libraries that define tree-sitter languages.")
 
 (defconst tree-sitter-langs--langs-with-deps
-  '(cpp typescript)
+  '(cpp typescript toml)
```
Comment (meain) [inline tree-sitter-langs-build.el:248]: I don't think we need this here. I'm not fully aware of this side of the codebase, but let me know if you had some specific reason to add this.

**Comment by meain** [comment]
The CI ran into two other issues (https://github.com/emacs-tree-sitter/tree-sitter-langs/issues/154 and https://github.com/emacs-tree-sitter/tree-sitter-langs/issues/159) once the previous one was fixed, could you rebase it once more on top of current master.

File: tree-sitter-langs-build.el (hunk 2 above)
Comment (mishazharov) [reply tree-sitter-langs-build.el:248]: I think if a treesitter repo is present in this list, then during the build `npm install` is executed to install any deps for that treesitter repo. I tried without this and it didn't seem to work (the build failed)

**Pull request merged**